engine/src/test/java/org/camunda/bpm/engine/test/standalone/history/HistoricTaskInstanceQueryTest.java

```diff
@@ -129,6 +129,29 @@ public void testProcessVariableValueLike() throws Exception {
     } catch (final ProcessEngineException e) {/*OK*/}
   }
 
+  @Deployment(resources = "org/camunda/bpm/engine/test/api/oneTaskProcess.bpmn20.xml")
+  @Test
+  public void testProcessVariableValueNotLike() throws Exception {
+    runtimeService.startProcessInstanceByKey("oneTaskProcess",
+            Collections.<String, Object>singletonMap("requester", "vahid alizadeh"));
+
+    assertEquals(0, historyService.createHistoricTaskInstanceQuery().processVariableValueNotLike("requester", "vahid%").count());
+    assertEquals(0, historyService.createHistoricTaskInstanceQuery().processVariableValueNotLike("requester", "%alizadeh").count());
+    assertEquals(0, historyService.createHistoricTaskInstanceQuery().processVariableValueNotLike("requester", "%ali%").count());
+
+    assertEquals(1, historyService.createHistoricTaskInstanceQuery().processVariableValueNotLike("requester", "requester%").count());
+    assertEquals(1, historyService.createHistoricTaskInstanceQuery().processVariableValueNotLike("requester", "%ali").count());
+
+    assertEquals(1, historyService.createHistoricTaskInstanceQuery().processVariableValueNotLike("requester", "vahid").count());
+    assertEquals(0, historyService.createHistoricTaskInstanceQuery().processVariableValueNotLike("nonExistingVar", "string%").count());
+
+    // test with null value
+    try {
+      historyService.createHistoricTaskInstanceQuery().processVariableValueNotLike("requester", null).count();
+      fail("expected exception");
+    } catch (final ProcessEngineException e) {/*OK*/}
```

Comment:
Same as the previous comment, we can refactor this code into an AssertJ statement like here: https://github.com/camunda/camunda-bpm-platform/blob/master/engine/src/test/java/org/camunda/bpm/engine/test/api/runtime/RuntimeServiceAsyncOperationsTest.java#L174-L176